src/miner.cpp

```diff
@@ -60,11 +60,12 @@ BlockAssembler::Options::Options() {
     nBlockMaxSize = DEFAULT_BLOCK_MAX_SIZE;
 }
 
-BlockAssembler::BlockAssembler(CChainState& chainstate, CEvoDB& evoDb, CMNHFManager& mnhfman, LLMQContext& llmq_ctx,
+BlockAssembler::BlockAssembler(CChainState& chainstate, CCreditPoolManager& cpoolman, CEvoDB& evoDb, CMNHFManager& mnhfman, LLMQContext& llmq_ctx,
                                MNSubsidyAgent& mn_subsidy, const CTxMemPool& mempool, const CChainParams& params, const Options& options) :
       chainparams(params),
       m_mempool(mempool),
       m_chainstate(chainstate),
+      m_cpoolman(cpoolman),
```

Comment:
with credit pool here that's getting even worse. Need to think how to wrap things up here. Credit pool and mnhfman and mn_subsidy - that's a things that should be inside m_chainstate, as by design and by idea.